### PR TITLE
Bug 1924440: CVE-2021-3121 gogo/protobuf lacks certain index validation

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR addresses CVE-2021-3121 gogo/protobuf lacks certain index validation.
The upstream fix in kubernetes is: kubernetes/kubernetes#98477

Resolves CVE:
cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-3121

Depends on: #650

/cc @blockloop
/assign @ewolinetz

### Links
- Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1924440

### CODE REVIEW

**Comment by jcantrill** [comment]
@periklis please rebase.  I'm hoping the recent changes to the image refs will unblock the failing test

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/elasticsearch-operator
 
-go 1.14
+go 1.15
```
Comment (ewolinetz) [inline go.mod:3]: is this required as part of the CVE fix?

Comment (periklis) [reply go.mod:3]: Nope this one slipped in.

**Comment by ewolinetz** [comment]
@periklis why is this cve fix dependent on the vendor dir being cleaned up?

**Comment by ewolinetz** [comment]
/approve

but please fix up the dependency for this

**Comment by periklis** [comment]
@blockloop PTAL

**Pull request merged**